### ISSUE TITLE
Upgrade runs successful but reports it failed

### ISSUE THREAD

**Issue by Martin-Weiss** [SUSE/suse-migration-services#89]
We did the upgrade and it seems it went through successful but at the end it reports upgrade failed and adds this to the /etc/issue.

Here the zypper.log and the distro_migration.log.
[ses6-up-20190522-logs.zip](https://github.com/SUSE/suse-migration-services/files/3207779/ses6-up-20190522-logs.zip)

**Comment by tserong** [comment]
Note: this was done with suse-migration-services modified to use `zypper dup` as follows:

```
--- suse_migration_services-0.5.24.orig/suse_migration_services/units/migrate.py
+++ suse_migration_services-0.5.24/suse_migration_services/units/migrate.py
@@ -41,15 +41,16 @@ def main():
         log.info('Running migrate service')
         bash_command = ' '.join(
             [
-                'zypper', 'migration',
+                'zypper', 
                 '--non-interactive',
                 '--gpg-auto-import-keys',
-                '--no-selfupdate',
+                #'--no-selfupdate',
+                #'--strict-errors-dist-migration',
+                #'--product', get_migration_product(),
+                '--root', root_path,
+		'dup',
                 '--auto-agree-with-licenses',
-                '--strict-errors-dist-migration',
                 '--replacefiles',
-                '--product', get_migration_product(),
-                '--root', root_path,
                 '&>>', Defaults.get_migration_log_file()
             ]
         )
```

**Comment by schaefi** [comment]
Well according to the log file there were several problems zypper had with that dup call, just to mention one: ```warning: %post(plymouth-branding-SLE-15-12.36.noarch) scriptlet failed, exit status 1```.

Jesus has code in the zypper migration plugin that deals with errors from zypper in a way that not all of them are handled as real errors and we continue with the migration as successful operation. All of that has been added as part of our development and also the image and packages we have released contains that fix.

Now you are telling me that you have modified something and that you are using some modified image build and that it has failed. So please let me clarify that our development model will not be able to work with that because we don't know anything about your changes nor which base image build you have used to apply the changes.

That makes the collaboration pretty difficult. Don't get me wrong, your contributions are really welcome but we should find a way that you participate in the workflow and also benefit from the changes we made. I remind you the network patches which lead to a problem on your side because you were based on an old image build.

So here is how it should work for all of us:

1. Branch the development image build:

    ```
    $ osc branch Devel:PubCloud/SLES12-SP4-SLES15-Migration
    ```

    If you need custom versions of a package to build here, e.g suse-migration-services
    just add the branch repo to the XML description and it gets picked up

2. Branch the development suse-migration-services package

    ```
    $ osc branch Devel:PubCloud/suse-migration-services
    ```

3.  With any issue let us know about your branch(s). Please note that regarding the distribution
     migration system all package and image builds for development are located in Devel:PubCloud

4. If you work on code from suse-migration-services create pull requests for review here, such that we know what you are doing. A code or patch paste in an issue is really hard to work with. Thanks

If you follow those steps we all can work together and everybody has a chance to look at everybody's code and we can also reproduce the show.

I hope you don't mind but I'm closing this because I think with the image build from Devel the issue might already been fixed due to the exit code handling Jesus added to the zypper plugin. If this is not the case please follow the suggested path or let us continue fixing the already open issue for this feature.

This brings me to my last point. You already opened an issue regarding the zypper dup feature. This means we plan for it and will also work on it. Now it seems in your team (SES) people are also working on it. This doesn't make sense and is waste of time and resources.

So please let us plan if you wish to address the issue yourself I'm more than happy. If not we will do it.

Thanks

**Comment by schaefi** [comment]
Closed on behalf of #72

**Comment by tserong** [comment]
Please accept my apologies, we're not trying to be difficult :-)  The patch I pasted in here was for reference, just so it was clear that we *had* done something different. That was a deliberately minimal hack to make it use `zypper dup`, in order to try `zypper dup` against SES6 + SLE15SP1; I didn't open a PR for this change as it was experimental, and the proper implementation was meant to be based on a config file as suggested in #72.
